git/odb/commit.go

```diff
@@ -0,0 +1,216 @@
+package odb
+
+import (
+	"bufio"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/git-lfs/git-lfs/errors"
+)
+
+// Signature represents a commit signature, which can represent either
+// committership or authorship of the commit that this signature belongs to. It
+// specifies a name, email, and time that the signature was created.
+type Signature struct {
+	// Name is the first and last name of the individual holding this
+	// signature.
+	Name string
+	// Email is the email address of the individual holding this signature.
+	Email string
+	// When is the instant in time when the signature was created.
+	When time.Time
+}
+
+const (
+	parseTimeZoneOnly  = "2006 -0700"
+	formatTimeZoneOnly = "-0700"
+)
+
+// String implements the fmt.Stringer interface and formats a Signature as
+// expected in the Git commit internal object format. For instance:
+//
+//  Taylor Blau <ttaylorr@github.com> 1494258422 -0600
+func (s *Signature) String() string {
+	at := s.When.Unix()
+	zone := s.When.Format(formatTimeZoneOnly)
+
+	return fmt.Sprintf("%s <%s> %d %s", s.Name, s.Email, at, zone)
+}
+
+var (
+	signatureNameRe  = regexp.MustCompile("^[^<]+")
+	signatureEmailRe = regexp.MustCompile("<(.*)>")
+	signatureTimeRe  = regexp.MustCompile("[-]?\\d+")
+)
+
+// ParseSignature parses a given string into a signature instance, returning any
+// error that it encounters along the way.
+//
+// ParseSignature expects the signature encoded in the given string to be
+// formatted correctly, and reproduce-able by the Signature.String() function
+// above.
+func ParseSignature(str string) (*Signature, error) {
+	name := signatureNameRe.FindString(str)
+	emailParts := signatureEmailRe.FindStringSubmatch(str)
+	timeParts := signatureTimeRe.FindAllStringSubmatch(str, 2)
```

Comment:
In git.git we have some gross heuristics to do our best with malformed entries. E.g., we parse the name/email left-to-right, but then parse the timestamp from the end, which leaves malformed or duplicated email bits in the middle as missing cruft. I don't know how thorough you'd want to be here. In fact, I'm not quite sure why you need to parse these ident lines in the first place. An LFS migration should be able to just treat them as opaque.